azurelinuxagent/ga/update.py

```diff
@@ -409,6 +413,21 @@ def run(self, debug=False):
         self._shutdown()
         sys.exit(0)
 
+    def _get_vm_size(self, protocol):
+        if self._vm_size is None:
```

Comment:
This check won't retry on next heartbeat if previous call sets to "unknown" due to failure.

Reply:
NM, I misunderstood. This looks good.